### ISSUE TITLE
False positive invalid-overriden-method with async generators

### ISSUE THREAD

**Issue by GoodWasHere** [PyCQA/pylint#3797]
### Steps to reproduce
1. Create a file `example.py`

```
"""Module docstring"""

class ClassA:
    """Class docstring"""
    def method(self):
        """Method docstring"""

    def method2(self):
        """Method docstring"""


class ClassB(ClassA):
    """Class docstring"""
    async def method(self):
        for i in range(3):
            yield i
```

2. Run `pylint example.py`

### Current behavior
`example.py:14:4: W0236: Method 'method' was expected to be 'non-async', found it instead as 'async' (invalid-overridden-method)`

### Expected behavior
No warnings in output

### pylint --version output
pylint 2.6.0
astroid 2.4.2
Python 3.7.4 (default, Jan 17 2020, 19:17:11)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

### Additional information
`method` returns async generator, not coroutine, and should be called sinchronously.

**Comment by PCManticore** [comment]
Thanks for the report. This behaviour is expected, `pylint` wants you to redefine a method according to how the method was defined in the base class.
